coreSDK/JavaScriptSDK/AppInsightsCore.ts

```diff
@@ -40,11 +39,33 @@ export class AppInsightsCore implements IAppInsightsCore {
         });        
 
         this._extensions = extensions.sort((extA, extB) => {
-            return extA.priority > extB.priority ? -1 : 1;
+            let typeExtA = typeof (<any>extA).processTelemetry;
+            let typeExtB = typeof (<any>extB).processTelemetry;
+            if (extA && typeExtA === 'function' && typeExtB === 'function') {
+                return (<any>extA).priority > (<any>extB).priority ? 1 : -1;
+            }
+
+            if (extA && 
+                typeExtA === 'function' && 
+                typeExtB !== 'function') {
+                // keep non telemetryplugin specific extensions at start
+                return 1;
+            }
+
+            if (extA && 
+                typeExtA !== 'function' && 
+                typeExtB === 'function') {
+                return -1;
+            }
         });
 
-        for (let idx = 0; idx < this._extensions.length - 2; idx++) {
-            this._extensions[idx].setNextPlugin(this._extensions[idx + 1]); // set next plugin
+        for (let idx = 0; idx < this._extensions.length - 1; idx++) {
```

Comment:
>- 1 [](start = 56, length = 3)

Are we doing -1 because we are assuming that last item is the channel? If so, can you add a comment noting this.

Reply:
done.